exp/api/v1alpha4/azuremanagedcontrolplane_webhook.go

```diff
@@ -281,3 +287,97 @@ func (r *AzureManagedControlPlane) validateSSHKey() error {
 
 	return nil
 }
+
+func (r *AzureManagedControlPlane) validateAadProfile() error {
+	if r.Spec.AADProfile != nil {
+		managedAad, err := r.validateManagedAadProfile()
+		if err != nil {
+			return err
+		}
+		legacyAad := r.validateLegacyAadProfile()
+		if managedAad && legacyAad {
+			return errors.New("conflicting values provided in AADProfile")
+		}
+	}
+	return nil
+}
+
+func (r *AzureManagedControlPlane) validateManagedAadProfile() (bool, error) {
+	if r.Spec.AADProfile.ManagedAAD != nil {
+		aad := r.Spec.AADProfile.ManagedAAD
+
+		if *aad.Managed && len(*aad.AdminGroupObjectIDs) == 0 {
+			return false, errors.New("require atleast one groupObjectID in Spec.AADProfile.ManagedAAD.AdminGroupObjectID")
+		}
+
+		if len(*aad.AdminGroupObjectIDs) != 0 && !*aad.Managed {
+			return false, errors.New("managed field has to be true to enable aad integration with aks")
+		}
+	}
+	return r.Spec.AADProfile.ManagedAAD != nil, nil
+}
+
+func (r *AzureManagedControlPlane) validateLegacyAadProfile() bool {
+	return r.Spec.AADProfile.LegacyAAD != nil
```

Comment:
is this missing some additional validation?

Reply:
Nope I guess I can remove the function here and directly do a check above. But this would be required only during an update now. Support during creation will be removed.